forge/ee/lib/gitops/index.js

```diff
@@ -0,0 +1,140 @@
+const { exec } = require('node:child_process')
+const fs = require('node:fs/promises')
+const os = require('node:os')
+const path = require('node:path')
+const { promisify } = require('node:util')
+const execPromised = promisify(exec)
+
+const axios = require('axios')
+
+module.exports.init = async function (app) {
+    // Set the git feature flag
+    app.config.features.register('gitIntegration', true, true)
+
+    /**
+     * Push a snapshot to a git repository
+     * @param {Object} repoOptions
+     * @param {String} repoOptions.token
+     * @param {String} repoOptions.url
+     * @param {String} repoOptions.branch
+     * @param {Object} snapshot
+     * @param {Object} options
+     * @param {Object} options.sourceObject what produced the snapshot
+     * @param {Object} options.user who triggered the pipeline
+     * @param {Object} options.pipeline details of the pipeline
+     */
+    async function pushToRepository (repoOptions, snapshot, options) {
+        let workingDir
+        try {
+            const token = repoOptions.token
+            const branch = repoOptions.branch || 'main'
+            if (!/^https:\/\/github.com/i.test(repoOptions.url)) {
+                throw new Error('Only GitHub repositories are supported')
+            }
+            const url = new URL(repoOptions.url)
+            url.username = 'x-access-token'
+            url.password = token
+
+            // 2. get user details so we can properly attribute the commit
+            let userDetails
+            try {
+                userDetails = await axios.get('https://api.github.com/user', {
+                    headers: {
+                        Accept: 'application/vnd.github+json',
+                        Authorization: `Bearer ${token}`,
+                        'X-GitHub-Api-Version': '2022-11-28'
+                    }
+                })
+            } catch (err) {
+                const result = new Error('Invalid git token')
+                result.code = 'invalid_token'
+                result.cause = err
+                throw result
+            }
+
+            const userGitName = userDetails.data.login
+            const userGitEmail = `${userDetails.data.id}+${userDetails.data.login}@users.noreply.github.com`
+            const author = `${userGitName} <${userGitEmail}>`.replace(/"/g, '\\"')
+            workingDir = await fs.mkdtemp(path.join(os.tmpdir(), 'flowfuse-git-repo-'))
```

Comment:
I thought we were going to run this in a directory of our choosing for the containers, not the default OS temp dir?

e.g. under the same `var` directory?

Reply:
This was a placeholder to get something done. Will add a config option for this (but use this as the fall back if not set)

Reply:
As discussed, we believe this is good as-is. It will default to `/tmp/` but can be overridden by `TMPDIR`, `TEMP`, `TMP` env vars (in that order)